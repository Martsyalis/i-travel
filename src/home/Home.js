import React, { PureComponent } from 'react';
import Feed from './Feed';
import { connect } from 'react-redux';
import { loadExpByUser } from '../experience/actions';
import { UpdateProfile } from '../auth/actions';

const defaultImg = 'http://www.beautythis.com/themes/comb/img/avatar/default-avatar-male_11.png';
class Home extends PureComponent {
  
  state={ shouldDisplay: false }


  handleUserUpdate = event =>{
    event.preventDefault();
    const form = event.target;
    const profile = new FormData(form);
    form.reset();
    this.props.UpdateProfile(profile);
  }

  render() {
    return (
      <div>
        <div>
          <img style={{ borderRadius: '50%', objectFit:'cover', width:'100px', height:'100px' }} alt='avatar' src={this.props.user.imageURI|| defaultImg}/>
          <h1>Hey {this.props.user.name}</h1>
          <button className="button" onClick={()=>{
            this.state.shouldDisplay
              ? this.setState({ shouldDisplay: false })
              : this.setState({ shouldDisplay: true });
          }}> EditProfile </button>
        </div>
        <div>
          { this.state.shouldDisplay &&
          <form onSubmit={this.handleUserUpdate}> 
            <input
              type="file"
              name="image"
              accept=".jpg, .jpeg, .png, .svg" 
            />
            <input name="name" defaultValue={this.props.user.name}/>
            <input name="email" defaultValue={this.props.user.email}/>
            <button type="submit">Add</button>
          </form>
          } 
        </div>
        <Feed/>
      </div>
    );
  }
} 
export default connect(
  state => ({ user: state.auth.user, exp: state.experiences }),
  { loadExpByUser, UpdateProfile }
)(Home);