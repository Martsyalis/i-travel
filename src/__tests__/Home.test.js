import React from 'react';

import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Home from '../home/Home';

describe('Home', () => {

  it('shows the Home component', () => {
    const wrapper = shallow(<Home />);
    console.log(toJSON(wrapper));
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});