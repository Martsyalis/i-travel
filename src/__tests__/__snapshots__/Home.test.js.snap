// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home shows the Home component 1`] = `
<div>
  <div>
    <img
      alt="avatar"
      src="http://www.beautythis.com/themes/comb/img/avatar/default-avatar-male_11.png"
      style={
        Object {
          "borderRadius": "50%",
          "height": "100px",
          "objectFit": "cover",
          "width": "100px",
        }
      }
    />
    <h1>
      Hello 
    </h1>
    <button
      className="button"
      onClick={[Function]}
    >
       EditProfile 
    </button>
  </div>
  <div />
  <Connect(Feed) />
</div>
`;
